test/e2e/tests/test_version.py

```diff
@@ -81,7 +81,9 @@ def lambda_function():
 @pytest.mark.canary
 class TestVersion:
     def test_smoke(self, lambda_client, lambda_function):
-        (_, function_resource) = lambda_function
+        (function_reference, function_resource) = lambda_function
+        fn_resource = k8s.get_resource(function_reference)
```

Comment:
Is this `fn_resource` actually different than `function_resource`? Any reason to not just use the same variable?

Reply:
Thanks for bringing this up. Have kept `fn_resource` and used it. Need to use `fn_resource` as `k8s.get_resource` is passing the whole function resource which can then be modified, which is required in our case.

Reply:
Update: have kept the `function_resource` which we are getting as second parameter to update the resource and used `k8s.wait_resource_consumed_by_controller` function to get the updated resource after making changes. This way it works too.